torchtune/models/clip/_component_builders.py

```diff
@@ -0,0 +1,74 @@
+from typing import List, Optional, Callable
+
+import torch
+from torchtune.modules import VisionTransformer, CLSProjection
+from torchtune.models.clip._position_embeddings import TokenPositionalEmbedding, TiledTokenPositionalEmbedding, TilePositionalEmbedding
+
+import logging
+
+logger = logging.getLogger(__name__)
+
+def clip_vision_encoder(
+    embed_dim: int,
+    num_layers: int,
+    num_heads: int,
+    act_layer: Callable,
+    indices_return_hidden: Optional[List[int]] = None,
+    output_cls_projection: bool = False,
+    tile_size: int = 512,
+    patch_size: int = 14,
+    max_num_tiles: int = 4,
+    mlp_ratio: float = 4.0,
+    in_channels: int = 3,
+    attn_dropout: float = 0.0,
+    norm_eps: float = 1e-5,
+    cls_output_dim: int = 512,
+) -> VisionTransformer:
+
+    patch_grid_size = tile_size // patch_size
+
+    cls_projection = CLSProjection(embed_dim=embed_dim, cls_output_dim=cls_output_dim) if output_cls_projection else None
+    
+    transformer_layer = torch.nn.TransformerEncoderLayer(
+        d_model=embed_dim, 
+        nhead=num_heads, 
+        dim_feedforward=int(mlp_ratio * embed_dim), 
+        dropout=attn_dropout, 
+        activation=act_layer, 
+        layer_norm_eps=norm_eps, 
+        batch_first=True, 
+        norm_first=True, 
+        bias=True)
+
+    # position embeddings
+    if max_num_tiles == 1:
+        logger.info("Found max_num_tiles=1. Setting tile_pos_embed to None and using only token_pos_embedding.")
```

Comment:
Similar comment here, I would remove the logging.

Reply:
I think it conveys important information. This behavior may be hidden from the user otherwise. They can also suppress INFO if they want to. Do you think that leaving the log harms the user experience? If you still think we should remove, just give me a thumbs up.

Reply:
(To explain the thumbs up) Mainly I think we will eventually provide some default builder where the user does not actually pass any arguments (think something like our existing `llama3_8b` function). In that case, `max_num_tiles` will not be changed by most users in practice (they would just use a different builder without passing it directly). So because of that I think the logging is more granular than what most people will care about and will just add extra noise.